Parse questions

jgoley opened the pull request:
- Adds `Question` class and refactors the meteor question creation method
- Adds a method on the form class to fetch its questions
- Cleans up things - removes mongo related stuff and mock data
- Refactors question removal
- Shares a collection between question edit and question list templates so when one is added to the Parse server, it appears reactively in the list.

We will need to add a story about question editing.

dan-nyanko commented:
@jgoley The first question added requires the page to be refreshed in order to be displayed within the list

dan-nyanko commented on app/client/questions/add_question.coffee:
```diff
@@ -108,18 +111,23 @@ Template.add_question.events
+          form.reset()
+          instance.choices.find().forEach ({_id})->
+            instance.choices.remove _id
+          instance.type.set 'inputText'
```
What's the purpose of defaulting the instance.type to 'inputText' upon successful creation of a question?

jgoley replied:
Not sure. I think @nathanathan implemented this. I suppose if a user is creating multiple questions of the same type, reseting the type would be frustrating.

dan-nyanko commented on app/client/questions/add_question.coffee:
```diff
@@ -108,18 +111,23 @@ Template.add_question.events
+          instance.choices.find().forEach ({_id})->
+            instance.choices.remove _id
+          instance.type.set 'inputText'
+          lastItemOrder = instance.questions.findOne({}, sort: {order: -1}).order
```
I believe the exception is thrown here due to questions is undefined on initial page load

jgoley replied:
Yep, just fixed it: https://github.com/ecohealthalliance/mobile-survey/blob/parse-questions/app/client/questions/add_question.coffee#L126-L128

dan-nyanko commented on app/client/questions/question.coffee:
```diff
@@ -0,0 +1,4 @@
+Template.question.helpers
+  questionTemplateName: ->
+    data = Template.currentData()
+    return "question.#{data.question_type}"
```
I'm not sure this template is ever rendered, but should this be camelCase?

jgoley replied:
I went ahead and removed this file and the related controller since its templates were replaced by this template: https://github.com/ecohealthalliance/mobile-survey/blob/pseudo-master/app/client/questions/add_question.jade

jgoley commented:
@dan-nyanko Thanks for taking a look. I fixed the issue causing the first question to not show in the list and the related exception.
I also removed the type reset on question creation.
